### PR TITLE
Add kustomization overlay: katib-openshift

### PR DESCRIPTION
**Pull request by maanur**
**What this PR does / why we need it**:
Current version of Katib have some issues and inconveniences when deployed to OpenShift, so I tried to introduce a workaround in form of kustomization overlay.

Workaround to #1512, but only for OpenShift 4.4+

**Special notes for your reviewer**:
Tested against OpenShift 4.6.21

**Release note**:
```release-note
NONE
```

### CODE REVIEW

**Comment by maanur** [comment]
@googlebot I signed it!

**Comment by maanur** [comment]
@googlebot I signed it!

File: manifests/v1beta1/installs/katib-standalone-openshift/kustomization.yml
Comment (andreyvelich) [inline manifests/v1beta1/installs/katib-standalone-openshift/kustomization.yml:43]: You don't need this RBAC. It is used for the cert-generator Job.
```suggestion
```

File: manifests/v1beta1/installs/katib-standalone-openshift/kustomization.yml
Comment (andreyvelich) [inline manifests/v1beta1/installs/katib-standalone-openshift/kustomization.yml:1]: Let's name this install: `katib-openshift` to be consistent with `katib-ibm`.
What do you think @maanur  ?

**Review by andreyvelich — COMMENTED** [review]
Thank you for this effort, this is very helpful!
I think we should wait until we merge this PR: https://github.com/kubeflow/katib/pull/1498, since we are making some changes to store Katib manifests.
After that, we can rebase this PR and merge it.

/cc @yanniszark @DavidSpek @johnugeorge @gaocegege

**Comment by DavidSpek** [comment]
@andreyvelich I'll work on my PR tomorrow, hopefully it will then be ready to merge. If after that any help is needed with this PR I'm happy to provide assistance.

**Comment by andreyvelich** [comment]
Thank you for your help @DavidSpek!

File: manifests/v1beta1/installs/katib-standalone-openshift/kustomization.yml
Comment (maanur) [reply manifests/v1beta1/installs/katib-standalone-openshift/kustomization.yml:1]: Sounds good

**Comment by andreyvelich** [comment]
@maanur I apologies for the delays on this PR.
Please, can you rebase and we can merge it ?

File: manifests/v1beta1/installs/katib-openshift/katib-config-patch.yaml
Comment (DavidSpek) [inline manifests/v1beta1/installs/katib-openshift/katib-config-patch.yaml:1]: This file should not be used anymore and as the image tags have been added to the controller deployment [here](https://github.com/kubeflow/katib/blob/master/manifests/v1beta1/components/controller/katib-config.yaml) instead.

File: manifests/v1beta1/installs/katib-openshift/kustomization.yml
Comment (DavidSpek) [inline manifests/v1beta1/installs/katib-openshift/kustomization.yml:54]: This patch can be removed as this file is not used anymore.

File: manifests/v1beta1/installs/katib-openshift/kustomization.yml
Comment (DavidSpek) [inline manifests/v1beta1/installs/katib-openshift/kustomization.yml:42]: This section should be replaced with:
```
- ../katib-standalone/
```
This way the Kustomize `load-restrictor` flag is not necessary and the manifest can be used with Kustomize v3 and v4 more easily. The other manifests have been updated in https://github.com/kubeflow/katib/pull/1498 to also use this style of referencing rather than linking to files directly.

File: manifests/v1beta1/installs/katib-openshift/kustomization.yml
Comment (DavidSpek) [inline manifests/v1beta1/installs/katib-openshift/kustomization.yml:52]: The cert-generator image is missing here which might be good to add. The manifests now also have the `newTag` under the `newName` to be more inline with the manifests in the other repositories. This is the `images` from katib-standalone:

```
images:
  - name: docker.io/kubeflowkatib/katib-controller
    newName: docker.io/kubeflowkatib/katib-controller
    newTag: latest
  - name: docker.io/kubeflowkatib/katib-db-manager
    newName: docker.io/kubeflowkatib/katib-db-manager
    newTag: latest
  - name: docker.io/kubeflowkatib/katib-ui
    newName: docker.io/kubeflowkatib/katib-ui
    newTag: latest
  - name: docker.io/kubeflowkatib/cert-generator
    newName: docker.io/kubeflowkatib/cert-generator
    newTag: latest
```

**Review by DavidSpek — COMMENTED** [review]
I've added some comment to help migrate this PR to implement the changes introduced in https://github.com/kubeflow/katib/pull/1498. After these changes this PR should be good to go.

File: manifests/v1beta1/installs/katib-openshift/kustomization.yml
Comment (DavidSpek) [reply manifests/v1beta1/installs/katib-openshift/kustomization.yml:42]: @maanur @andreyvelich Given that this PR was created to fix issues with the cert-generator (https://github.com/kubeflow/katib/issues/1512), I'm not sure if it might not be better to use `katib-cert-manager` instead. You would be better judges of that.

Comment (andreyvelich) [reply manifests/v1beta1/installs/katib-openshift/kustomization.yml:42]: I think it should be up to the OpenShift users. Maybe some of them don't want to use `cert-manager` to deploy Katib.

What do you think @maanur @nakfour @gaocegege @johnugeorge  ?

Comment (DavidSpek) [reply manifests/v1beta1/installs/katib-openshift/kustomization.yml:42]: That's also what I thought. An alternative would be for Katib to require cert-manager or have that be the default, but I can understand why this wouldn't be wanted as well so that is up to you guys.

Comment (maanur) [reply manifests/v1beta1/installs/katib-openshift/kustomization.yml:42]: This PR has nothing to do with cert-manager, relying on builtin OpenShift service controller instead.

File: manifests/v1beta1/installs/katib-openshift/kustomization.yml
Comment (maanur) [reply manifests/v1beta1/installs/katib-openshift/kustomization.yml:52]: cert-generator is not needed here

Comment (DavidSpek) [reply manifests/v1beta1/installs/katib-openshift/kustomization.yml:52]: Then I will update my recommendation above.

File: manifests/v1beta1/installs/katib-openshift/kustomization.yml
Comment (DavidSpek) [reply manifests/v1beta1/installs/katib-openshift/kustomization.yml:42]: @maanur If you are not relying on the cert-generator job you should change the `resources:` section to the following:

```yaml
# Namespace.
  - ../../components/namespace
  # Katib controller.
  - ../../components/controller/
  # Katib CRDs.
  - ../../components/crd/
  # Katib DB manager.
  - ../../components/db-manager/
  # Katib DB mysql.
  - ../../components/mysql/
  # Katib UI.
  - ../../components/ui/
  # Katib webhooks.
  - ../../components/webhook/
```

**Comment by maanur** [comment]
Manually tested against OpenShift version 4.6.21

**Comment by DavidSpek** [comment]
@maanur Looks good to me.
/lgtm

File: manifests/v1beta1/installs/katib-openshift/patch-pvc-unset-storageClass.yaml
Comment (andreyvelich) [inline manifests/v1beta1/installs/katib-openshift/patch-pvc-unset-storageClass.yaml:1]: Please, can you move all patches under `katib-openshift/patches` like for other installs: https://github.com/kubeflow/katib/tree/83678f1c3dad9bf648c7c0fb74a6ebd7323c3d3f/manifests/v1beta1/installs/katib-with-kubeflow or https://github.com/kubeflow/katib/tree/83678f1c3dad9bf648c7c0fb74a6ebd7323c3d3f/manifests/v1beta1/installs/katib-cert-manager

File: manifests/v1beta1/installs/katib-openshift/kustomization.yml
Comment (andreyvelich) [inline manifests/v1beta1/installs/katib-openshift/kustomization.yml:1]: Let's also update the release script: https://github.com/kubeflow/katib/blob/master/scripts/v1beta1/release.sh#L77.
Then, we will update the `katib-openshift` install images for the release branch also.

**Review by andreyvelich — COMMENTED** [review]
Thank you for this effort @maanur!
I left few comments from my side.

It would be great if you could also left your review @yanniszark @nakfour @gaocegege @johnugeorge

**Comment by andreyvelich** [comment]
Thank you for the changes @maanur.
/lgtm
/hold for the review from others.

**Review by andreyvelich — APPROVED** [review]
I think we can merge it.
Thank you for this effort @maanur!
/hold cancel
/approve

**Pull request merged**
